examples/example.hpp

```diff
@@ -1072,13 +1072,14 @@ void register_glfw_callbacks(window& app, glfw_state& app_state)
 }
 
 // Find devices with specified streams
-bool device_with_streams(std::vector <rs2_stream> types, std::string& out_serial)
+bool device_with_streams(std::vector <rs2_stream> stream_requests, std::string& out_serial)
 {
     rs2::context ctx;
-    for (auto type : types)
+    auto devs = ctx.query_devices();
+    for (auto type : stream_requests)
     {
         bool stream_found = false;
-        for (auto dev : ctx.query_devices())
+        for (auto& dev : devs)
```

Comment:
The check must be reset between iterations and counted per device to avoid false positives